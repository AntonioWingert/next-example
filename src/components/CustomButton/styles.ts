import tw from 'tailwind-styled-components'
import { CustomButtonProps } from '.'
import theme from '@/styles/theme'

export const Container = tw.button<CustomButtonProps>`
  ${({ $variable }) => $variable && theme['background-colors'][$variable]}
  
  rounded
  p-2
`
