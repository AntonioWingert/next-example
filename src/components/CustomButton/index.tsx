import React from 'react';

import * as S from './styles';

export type CustomButtonProps = {
  children: React.ReactNode;
  $variable?: 'danger' | 'success' | 'primary' | 'secondary';
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

const CustomButton = ({
  children,
  $variable = 'primary',
  ...props
}: CustomButtonProps) => {
  return (
    <S.Container $variable={$variable} {...props}>
      {children}
    </S.Container>
  );
};

export default CustomButton;
