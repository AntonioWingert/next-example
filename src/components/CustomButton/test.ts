import { render, screen } from '@testing-library/react';
import CustomButton from '.';

describe('<CustomButton />', () => {
  it('should render the button with all props passed', () => {
    const { container } = render(
      CustomButton({ children: 'Test', $variable: 'primary' }),
    );
    expect(screen.getByRole('button', { name: /Test/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the button with default props', () => {
    const { container } = render(CustomButton({ children: 'Test' }));
    expect(screen.getByRole('button', { name: /Test/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
