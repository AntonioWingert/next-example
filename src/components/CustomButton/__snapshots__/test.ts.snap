// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> should render the button with all props passed 1`] = `
<button
  class="bg-blue-200 rounded p-2"
>
  Test
</button>
`;

exports[`<CustomButton /> should render the button with default props 1`] = `
<button
  class="bg-blue-200 rounded p-2"
>
  Test
</button>
`;
